Auth repo-relative headers

paulb777 opened the pull request:
Public headers use module syntax and others are repo-relative.

Repo relative headers is the recommended approach at https://github.com/firebase/firebase-ios-sdk/blob/master/AddNewPod.md.  It makes projects more scalable since there's no way for an import to be ambiguous.  It makes managing github to g3 easier because the BUILD doesn't have to be updated when new headers are added.  And additional include paths on the command line can cause build performance issues for dependencies.

#no-changelog

wilhuff commented on FirebaseAuth/Sources/Auth/FIRAuth.m:
```diff
@@ -32,57 +32,57 @@
-#import "FIRVerifyPhoneNumberResponse.h"
-#import "FirebaseAuth.h"
+#import <FirebaseAuth/FIRAuthSettings.h>
+#import <FirebaseAuth/FirebaseAuth.h>
```
Isn't this the umbrella header? If so, doesn't it already include `FIRAuthSettings.h`?

Secondarily, shouldn't these be grouped with the other framework imports?

wilhuff commented on FirebaseAuth/Sources/AuthProvider/Email/FIREmailPasswordAuthCredential.m:
```diff
@@ -14,11 +14,11 @@
-#import "FIRAuthExceptionUtils.h"
-#import "FIREmailAuthProvider.h"
-#import "FIRVerifyAssertionRequest.h"
+#import <FirebaseAuth/FIREmailAuthProvider.h>
```
I'm surprised the linter isn't complaining about this. Headers that look like system headers should get their own group, no?

Here and throughout if you choose to take action.

wilhuff commented on FirebaseAuth/Tests/Unit/FIRAuthBackendRPCImplementationTests.m:
```diff
@@ -16,13 +16,13 @@
-#import "FIRAuthRPCResponse.h"
-#import "FIRAuthRequestConfiguration.h"
 #import "FIRFakeBackendRPCIssuer.h"
+#import "FirebaseAuth/Sources/Backend/FIRAuthBackend.h"
```
Repo-relative import for `FIRFakeBackendRPCIssuer.h`?

wilhuff replied:
There are still many instances of this file being unqualified. This should find them all:

```
grep -r '#import "[^/]*"$' FirebaseAuth/Tests/Unit
```

paulb777 replied:
Thanks for the grep command. The last commit should get the rest.

wilhuff commented on FirebaseAuth/Tests/Unit/FIRAuthNotificationManagerTests.m:
```diff
@@ -17,9 +17,9 @@
 #import <XCTest/XCTest.h>
 
```
XCTest and OCMock should group together, no?

wilhuff commented on FirebaseAuth/Tests/Unit/FIRPhoneAuthProviderTests.m:
```diff
@@ -17,33 +17,33 @@
 #import "OCMStubRecorder+FIRAuthUnitTests.h"
-#import "Phone/FIRPhoneAuthCredential_Internal.h"
 
 #import <SafariServices/SafariServices.h>
```
Framework imports above regular imports.

paulb777 commented:
Thanks @wilhuff. I'd prefer to have this PR be a pure translation to repo-relative headers along with subsequent style sorting, and g3 synchronization - and address your comments in the next PR

paulb777 commented:
@wilhuff Running clang-format with `IncludeBlocks: Regroup` addresses most of the concerns. I also made the unit test headers repo-relative and removed the redundant import in FirebaseAuth.m.

wilhuff commented on FirebaseAuth/Tests/Sample/ApiTests/FIRAuthApiTestsBase.h:
```diff
@@ -18,7 +18,7 @@
 #import <FirebaseCore/FIRApp.h>
 #import "AuthCredentials.h"
-#import "FirebaseAuth.h"
+#import <FirebaseAuth/FirebaseAuth.h>
```
`Tests/Sample` still seems unformatted, though I suppose that's specifically because of an exclusion in style.sh?

paulb777 replied:
Right.
